packages/uni_app/lib/controller/fetchers/course_units_fetcher/course_units_info_fetcher.dart

```diff
@@ -14,31 +14,37 @@ class CourseUnitsInfoFetcher implements SessionDependantFetcher {
     return NetworkRouter.getBaseUrlsFromSession(session).toList();
   }
 
-  Future<CourseUnitSheet> fetchCourseUnitSheet(
-    Session session,
-    int occurrId,
-  ) async {
-    // if course unit is not from the main faculty, Sigarra redirects
-    final url = '${getEndpoints(session)[0]}ucurr_geral.ficha_uc_view';
-    final response = await NetworkRouter.getWithCookies(
-      url,
-      {'pv_ocorrencia_id': occurrId.toString()},
-      session,
-    );
-    return parseCourseUnitSheet(response);
-  }
-
   Future<Sheet> fetchSheet(
     Session session,
     int occurId,
   ) async {
-    final url = '${getEndpoints(session)[0]}mob_ucurr_geral.perfil';
-    final response = await NetworkRouter.getWithCookies(
-      url,
-      {'pv_ocorrencia_id': occurId.toString()},
-      session,
+    final responses = await Future.wait(
+      getEndpoints(session)
+          .map((endpoint) => '$endpoint' 'mob_ucurr_geral.perfil')
+          .map(
+            (url) => NetworkRouter.getWithCookies(
+              url,
+              {'pv_ocorrencia_id': occurId.toString()},
+              session,
+            ),
+          ),
```

Comment:
You are missing the `catchError` call. If responses throw an HTTP 404 error, for instance, the Future.wait call with throw, whereas with `catchError((_) => null)`, null is returned and no error is thrown

Reply:
Thanks for pointing that out! I've corrected it :)